shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -468,6 +469,14 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  SkCanvas* background_canvas) {
+  if (merge_threads_) {
+    // Threads are about to be merged, we drop everything from this frame
+    // and possibly resubmit the same layer tree in the next frame.
+    picture_recorders_.clear();
+    composition_order_.clear();
+    return true;
+  }
+
   DisposeViews();
```

Comment:
you might want to run `DisposeViews` as well

Reply:
Also, RemoveUnusedLayers();

Reply:
I should clean this up, so there's a single method.

Reply:
Actually, we can't run `DisposeViews` here.
When we at the frame before thread merging, we know that the frame is running on the GPU thread.
DisposeViews has a bunch of UIKit operations that should not be run on the GPU thread.
We basically bail here and let the resubmit takes care of everything.

Reply:
I see. what about adding a two `FML_CHECK`s. One for the GPU thread within this if block, and one for the platform thread right before `DisposeViews`.  I think that would help understanding the execution.

Reply:
SG

Reply:
actually, disposeview might be called on the gpu as well. but only when the there are no platform views